core/sdk-core/src/main/java/software/amazon/awssdk/core/internal/async/SplittingTransformer.java

```diff
@@ -96,51 +99,145 @@ public class SplittingTransformer<ResponseT, ResultT> implements SdkPublisher<As
 
     /**
      * The amount requested by the downstream subscriber that is still left to fulfill. Updated. when the
-     * {@link Subscription#request(long) request} method is called on the downstream subscriber's subscription.
-     * Corresponds to the number of {@code AsyncResponseTransformer} that will be published to the downstream subscriber.
+     * {@link Subscription#request(long) request} method is called on the downstream subscriber's subscription. Corresponds to the
+     * number of {@code AsyncResponseTransformer} that will be published to the downstream subscriber.
      */
     private final AtomicLong outstandingDemand = new AtomicLong(0);
 
+    /**
+     * tracks if the publisher is currently sending AsyncResponseTransformer to the downstreamSubscriber
```

Comment:
Maybe something like `This flag stops the publisher from emit a new transformer while another transformer is currently in flight`